packages/tldraw/src/state/TldrawApp.ts

```diff
@@ -3056,6 +3079,13 @@ export class TldrawApp extends StateManager<TDSnapshot> {
   }
 
   onPointerDown: TLPointerEventHandler = (info, e) => {
+    if (e.buttons === 4) {
+      this.setStatus(Status.MiddleWheelPanning)
+      return
+    }
+    if (this.status === Status.SpacePanning) {
```

Comment:
This used to go to `idle` for 1 tick and then straight back to `spacePanning`. Prevented that because it caused some flickering while the rectangle tool was selected. Do you expect this change to cause any issues?